OrthoEvol/Orthologs/Blast/comparative_genetics.py

```diff
@@ -73,34 +78,62 @@ def __init__(self, project=None, project_path=os.getcwd(), acc_file=None,
         self.__pre_blast = pre_blast
         self.__post_blast = post_blast
         self.__taxon_filename = taxon_file
+        self.acc_file = acc_file
         self.acc_csv_filename = acc_file
+
+        # Initialize variables
+        self.project_path = project_path
         self.project = project
+        self.species = 'Homo_sapiens'
+        self.taxon_file = taxon_file
+        self.go_list = None
+        self.proj_mana = proj_mana
+        self.copy_from_package = kwargs['copy_from_package']
 
         # Initialize Logging
         self.get_time = time.time
         self.sep = 50*'*'
 
         # Initialize Utilities
         self.blast_utils = FullUtilities()
-
-        if project_path and project:
+        if self.project_path and self.project:
             self.project_path = Path(project_path) / Path(project)
+        elif self.project and not self.project_path:
+            self.project_path = self.project
+        # If user does not want to use a project name, create one anyway.
+        elif not self.project and not self.project_path:
+            four_ints = random.sample(range(1, 9), 4)  # 4 random integers
+            four_ints_str = ''.join(str(e) for e in four_ints)
+            self.project = "orthoevol" + four_ints_str  # New project name
+            self.project_path = os.getcwd()
+        # TODO: Add ability to use an existing project or project path.
+        elif self.project_path and not self.project:
+            raise NotImplementedError
+
+        self.blastn_log.debug('Project name: %s' % self.project)
+        self.blastn_log.debug('Project path: %s' % self.project_path)
 
         # Configuration of class attributes.
-        add_self = self.blast_utils.attribute_config(self, composer=proj_mana, checker=ProjectManagement, project=project, project_path=project_path)
+        add_self = self.blast_utils.attribute_config(cls=self,
+                                                     composer=proj_mana,
+                                                     checker=ProjectManagement,
+                                                     project=project,
+                                                     project_path=project_path)
         for variable, attribute in add_self.__dict__.items():
             setattr(self, variable, attribute)
 
         # Handle the taxon_id file and blast query
-        if taxon_file is not None:
+        if self.taxon_file is not None:
```

Comment:
Update this after reducing to one class variable (self.__taxon_filename or self.taxon_file.